pkg/service/restriction_service.go

```diff
@@ -0,0 +1,80 @@
+package gcpservice
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/kiwigrid/gcp-serviceaccount-controller/pkg/apis/gcp/v1beta1"
+	"regexp"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type RestrictionService struct {
+	log            logr.Logger
+	resolveService RestrictionResolveService
+}
+
+func NewRestrictionService(restrictionResolveService RestrictionResolveService) *RestrictionService {
+	return &RestrictionService{
+		log:            logf.Log.WithName("restrictionservice"),
+		resolveService: restrictionResolveService}
+}
+
+func (r *RestrictionService) CheckNamespaceHasRights(namespace string, resources []v1beta1.GcpRoleBindings) (bool, error) {
+	restriction, err := r.resolveService.CheckNamespaceHasRights(namespace)
+	if err != nil {
+		return false, err
+	}
+	//check if we find a match for each entry
+	for _, res := range resources {
+		if res.Resource != "" {
+			find, binding := getMatchingResource(restriction, res.Resource)
+			if !find {
+				return false, nil
+			}
+			return checkAllRolesMatch(binding, res.Roles), nil
+		}
+
+	}
+	return false, nil
+}
+
+func checkAllRolesMatch(binding *v1beta1.GcpRestrictionRoleBinding, roles []string) bool {
+	for _, role := range roles {
+		found := false
+		for _, check := range binding.Roles {
+			if checkIfMatch(check, role) {
+				found = true
+				break
+			}
+		}
+		if !found {
```

Comment:
it's superfluous, you can return `found` immediately (as it's initialized with `false` in line 42)